documentation/internal/resource-manager-active-directory.md

```diff
@@ -0,0 +1,292 @@
+
+In the future, it should be determined whether or not the `ResourceManagerClient` can be used to make the subscription level calls as it has two functions, `CreateOrUpdate()` and `CreateOrUpdateById()`, which should be able to perform the same steps outlined above.
+
+**Set-AzResource**
+
+Also like the `New-AzResource` cmdlet, the `Set-AzResource` cmdlet only uses the generic `HttpClient` to make calls to the server to update a resource. This cmdlet gives users the ability to perform either a `PUT` or `PATCH` operation on the server to update the given resource. If the user provides the identity properties of a resource as parameters, they can perform a `PATCH` operation by providing the `-UsePatchSemantics` switch, providing a value for the `-Tag` or `-Sku` parameters, or not providing a value for the `-Plan`, `-Properties` and `-Kind` parameters. If the user doesn't do [any of the previous three](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/SetAzureResourceCmdlet.cs#L226-L229), than the cmdlet will perform a `PUT` operation.
+
+If the user provides a `PSResource` object to the `-InputObject` parameter, then the cmdlet will push the value of the properties on the object to the server, [unless overwritten by a parameter provided on the command line](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/SetAzureResourceCmdlet.cs#L94-L102). For example, if the user provides a `PSResource` for `-InputObject` and a new `Hashtable` for `-Tag`, then the tags defined on the given resource will be overwritten with whatever is in the `Hashtable` provided, and all other properties on the `PSResource` will be pushed to the server.
+
+#### Piping Scenarios
+
+```powershell
+# Update the tags with a new key-value pair for all resources
+Get-AzResource @FilterParameters | ForEach-Object { $_.Tags.Add("SampleKey", "SampleValue") } | Set-AzResource -Force
+
+# Update a single property on a resource
+$Resource = Get-AzResource -ResourceGroupName MyResourceGroup -ResourceType Microsoft.Web/sites -Name MySite
+$Resource.Properties.Enabled = "False"
+$Resource | Set-AzResource -Force
+```
+
+## `Resources` project
+
+The `Resources` project contains all of the Active Directory, Authorization, and Management Groups cmdlets.
+
+### Active Directory Applications
+
+From the [_Application and service principal objects in Azure Active Directory_](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Test-AzResourceGroupDeployment.md) document, an Azure AD application is defined as the following:
```

Comment:
Excuse me, I think the link is not about _Application and service principal objects in Azure Active Directory_

Reply:
@wyunchi-ms yep, you're right, looks like a copy-paste error 😁